### PR TITLE
Update the user guide to use channel_axis rather than multichannel

### LINKED ISSUE #5554: Update user guide to reflect channel_axis vs. multichannel
## Description

The user guide has not been updated since the move from `multichannel` to `channel_axis`. We need to update sections like the following one: https://scikit-image.org/docs/stable/user_guide/numpy_images.html#coordinate-conventions

### PR DESCRIPTION
**Pull request by grlee77**
## Description

closes #5554

Updates the user guide to better describe the more flexible channel support via `channel_axis`.  Also, avoids use of the deprecated `multichannel` argument in examples.

The description of `img_as_float` was also updated slightly to indicate its current behavior (no longer automatically rescales floating point inputs and preserves floating point precision).

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: doc/source/user_guide/numpy_images.rst [hunk 1]
```diff
@@ -147,26 +152,35 @@ denote standard Cartesian coordinates, where ``x`` is the horizontal coordinate,
...
   2D multichannel (eg. RGB)   (row, col, ch)
   3D grayscale                (pln, row, col)
   3D multichannel             (pln, row, col, ch)
-  =========================   ========================================
+  =========================   =============================
 
+Note that the position of ``ch`` is controlled by the ``channel_axis``
+argument.
+
+|
```
Comment (mkcor) [inline doc/source/user_guide/numpy_images.rst:183]: Is this RST syntax for a note?

File: doc/source/user_guide/numpy_images.rst [hunk 2]
```diff
@@ -147,26 +152,35 @@ denote standard Cartesian coordinates, where ``x`` is the horizontal coordinate,
...
+In the case of multichannel images, any dimension can be used for color
+channels, and is denoted by ``channel`` or ``ch``. Prior to scikit-image 0.19,
+the channels axis was always last, but in the current release the channel
+position can be specified by a ``channel_axis`` argument. Functions that
+require multichannel data default to ``channel_axis = -1``. Otherwise,
+functions default to ``channel_axis = None``, indicating that no axis is
+assumed to correspond to channels.
 
 Finally, for volumetric (3D) images, such as videos, magnetic resonance imaging
 (MRI) scans, confocal microscopy, etc. we refer to the leading dimension
```
Comment (mkcor) [inline doc/source/user_guide/numpy_images.rst:164]:
```suggestion
(MRI) scans, confocal microscopy, etc., we refer to the leading dimension
```

File: doc/source/user_guide/numpy_images.rst [hunk 3]
```diff
@@ -147,26 +152,35 @@ denote standard Cartesian coordinates, where ``x`` is the horizontal coordinate,
 ``y`` - the vertical one, and the origin is at the bottom left
 (Matplotlib axes, for example, use this convention).
 
-In the case of multichannel images, the last dimension is used for color channels
-and is denoted by ``channel`` or ``ch``.
+In the case of multichannel images, any dimension can be used for color
+channels, and is denoted by ``channel`` or ``ch``. Prior to scikit-image 0.19,
+the channels axis was always last, but in the current release the channel
+position can be specified by a ``channel_axis`` argument. Functions that
```
Comment (mkcor) [inline doc/source/user_guide/numpy_images.rst:158]: In the above, is it clear to the newbie reader that dimension, axis, and position mean the same thing?

File: doc/source/glossary.md [hunk 4]
```diff
@@ -11,6 +11,17 @@ array
     correspond to spatial dimensions of the image, and color channels for
     color images. See {ref}`numpy`. 
 
+channel
+    Typically used to refer to a single color channel in a color image. RGBA
+    images have an additional alpha (transparency) channel. Functions use a
+    ``channel_axis`` argument to specify which axis of an array corresponds
+    to channels. Images without channels are indicated via
+    ``channel_axis=None``. Aside from the functions in ``skimage.color``, most
+    functions with a channel_axis argument just apply the same operation
```
Comment (mkcor) [inline doc/source/glossary.md:20]:
```suggestion
    functions with a ``channel_axis`` argument just apply the same operation
```
Maybe?

**Review by mkcor — COMMENTED** [review]
Thanks for taking care of this!

File: doc/source/user_guide/numpy_images.rst [hunk 5]
```diff
@@ -250,11 +264,11 @@ shape (t, pln, row, col, ch)::
...
-.. table:: Addendum to dimension names and orders in scikit-image
+.. table:: *Addendum to dimension names and orders in scikit-image*
 
-  ========================   ========================================
+  ========================   =========================================
   Image type                 coordinates
-  ========================   ========================================
+  ========================   =========================================
   2D color video             (t, row, col, ch)
   3D multichannel video      (t, pln, row, col, ch)
```
[the remark this answers is not in the document]
Comment (grlee77) [reply doc/source/user_guide/numpy_images.rst:273]: yes, that sounds like a good change.

File: doc/source/user_guide/numpy_images.rst (hunk 3 above)
Comment (grlee77) [reply doc/source/user_guide/numpy_images.rst:158]: yeah, let me rephrase this a bit and remove use of "position", at least. Hopefully the user will be able to relate "dimension" and "axis" as equivalent terms?  Maybe the first sentence here can make this clearer by reading adding a mention of axis as in:

```
... any dimension (array axis) can be used ...
```

File: doc/source/user_guide/numpy_images.rst (hunk 1 above)
Comment (grlee77) [reply doc/source/user_guide/numpy_images.rst:183]: I just forces a little extra whitespace in the document. I didn't see any command to add a table footnote, so I was trying to mess with the spacing so that it was clearer that the note above applies to the table.

File: doc/source/user_guide/numpy_images.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (grlee77) [reply doc/source/user_guide/numpy_images.rst:273]: The prior table used multichannel which I think is nice as it doesn't imply channels must be colors. I think "color video" is likely clearer than "multichannel video" here, though

File: doc/source/user_guide/numpy_images.rst (hunk 1 above)
Comment (grlee77) [reply doc/source/user_guide/numpy_images.rst:183]: This docs section describes use of pipe (|). The pipe symbol itself doesn't get rendered
https://bashtage.github.io/sphinx-material/rst-cheatsheet/rst-cheatsheet.html#blocks

https://stackoverflow.com/questions/14207327/how-to-insert-blank-line-using-restructuredtext-sphinx

File: doc/source/user_guide/numpy_images.rst (hunk 3 above)
Comment (mkcor) [reply doc/source/user_guide/numpy_images.rst]: Great rephrasing, thanks, @grlee77!

File: doc/source/user_guide/numpy_images.rst [hunk 6]
```diff
@@ -147,26 +152,35 @@ denote standard Cartesian coordinates, where ``x`` is the horizontal coordinate,
...
 (Matplotlib axes, for example, use this convention).
 
-In the case of multichannel images, the last dimension is used for color channels
-and is denoted by ``channel`` or ``ch``.
+In the case of multichannel images, any dimension (array axis) can be used for
+color channels, and is denoted by ``channel`` or ``ch``. Prior to scikit-image
+0.19, this channels dimension was always last, but in the current release the
+channel dimension can be specified by a ``channel_axis`` argument. Functions
+that require multichannel data default to ``channel_axis = -1``. Otherwise,
+functions default to ``channel_axis = None``, indicating that no axis is
```
Comment (mkcor) [inline doc/source/user_guide/numpy_images.rst:160]:
```suggestion
that require multichannel data default to ``channel_axis=-1``. Otherwise,
functions default to ``channel_axis=None``, indicating that no axis is
```
Just noticed this different convention wrt `doc/source/user_guide/data_types.rst` (showing ``preserve_range=False``). It's no big deal, but I guess the no-space version reflects better the practical use as a function argument?

File: doc/source/user_guide/numpy_images.rst [hunk 7]
```diff
@@ -147,26 +152,35 @@ denote standard Cartesian coordinates, where ``x`` is the horizontal coordinate,
 ``y`` - the vertical one, and the origin is at the bottom left
 (Matplotlib axes, for example, use this convention).
 
-In the case of multichannel images, the last dimension is used for color channels
-and is denoted by ``channel`` or ``ch``.
+In the case of multichannel images, any dimension (array axis) can be used for
+color channels, and is denoted by ``channel`` or ``ch``. Prior to scikit-image
+0.19, this channels dimension was always last, but in the current release the
```
Comment (mkcor) [inline doc/source/user_guide/numpy_images.rst:157]:
```suggestion
0.19, this channel dimension was always last, but in the current release the
```
Not wrong, but I realize the following line reads 'the channel dimension' (and it saves one character). Sorry, I'm the nitpickiest!! :innocent:

File: doc/source/user_guide/numpy_images.rst (hunk 1 above)
Comment (mkcor) [reply doc/source/user_guide/numpy_images.rst:183]: Ok, I didn't know, thanks for the tip!

**Comment by grlee77** [comment]
Updated. Thanks for the helpful suggestions, @mkcor.

**Comment by grlee77** [comment]
The `linux-cp3.8-pre` failure here is unrelated (appears to be an issue with a new `setuptools_scm` release)

**Review by mkcor — APPROVED** [review]
